### DISCUSSION TITLE
[Help Wanted] Porting MIUI support from latest MIUI Global ROM (original) to any other ROM (AOSP)

### DISCUSSION THREAD

**Discussion by joeyoropesa-dev** [reiryuki/Miui-Core-Magisk-Module#26] [Q&A]
I bought and rooted Xiaomi Mi 11 Lite 5G NE and I want from this device and latest MIUI version to copy all required files and folders from /system and other directories and to put it in right places in MiuiCore module to make latest MIUI Compatibility layer for any ROM that is not MIUI

But how can I do this without bootlooping an Samsung device?

What exactly files and folders I need to copy to the module that will provide full MIUI apps support (like Mi Account login to sync Mi Gallery private album and other photos from Mi services to Mi Gallery)

Is it maybe after all possible to port it (copy & paste into module) and to work or all modern libraries and core apps of MIUI will require for real an real MIUI system and it's components that cannot be copied

I hope that is possible and that I will get an list about what exactly I need to copy so that I can get latest MIUI apps to work (that includes list of MIUI Core apps and libraries that are required for an app like Mi Gallery with Mi Account from Settings to work as expected)

That's all I need to migrate from one device to another - thanks in advance 🥰

(I've moved here since using GitHub issues it seems it's not recommended for these kind of questions - as I saw that discussion was used by users already, I've decided to try it too)

**Comment by reiryuki** [comment]
Miui Core is not for Miui/HyperOS ROM.
MIUI Core is essentially just a porting tool to add missing classes and resources needed for porting MIUI/HyperOS apps.
Yeah, so this isn't a magic trick that can suddenly make MIUI apps work automatically in AOSP ROMs (except for a few simple, non-system apps).
I've already written all the explanations completely in the README.
For bootloop problems, that is the responsibility of yourself.
